notebook2/notebook/notebook.py

```diff
@@ -190,21 +215,26 @@ def root():
 
 
 @app.route('/new', methods=['GET'])
+@requires_auth()
 def new_get():
     pod_name = session.get('pod_name')
     svc_name = session.get('svc_name')
     if pod_name:
         delete_worker_pod(pod_name, svc_name)
-    session.clear()
+        del session['pod_name']
+
+    if svc_name is not None:
+        del session['svc_name']
```

Comment:
it's an invariant of the system that both pod_name and svc_name are set or both not set, so I'd include this under the previous if, but I don't much care since we're removing services shortly anyway.

Reply:
Yeah, it will go away. I see your point, you're right. In this PR it's gone: https://github.com/hail-is/hail/pull/5476